src/mlpack/methods/ann/loss_functions/log_cosh_loss.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file log_cosh_loss.hpp
+ * @author Kartik Dutt
+ *
+ * Definition of the Log-Hyperbolic-Cosine loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_LOG_COSH_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_LOG_COSH_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Log-Hyperbolic-Cosine loss function is often used to improve
+ * variational auto encoder. This function is the log of hyperbolic 
+ * cosine of difference between true values and predicted values.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+        typename InputDataType = arma::mat,
+        typename OutputDataType = arma::mat
+>
+class LogCoshLoss
+{
+ public:
+  /**
+   * Create the Log-Hyperbolic-Cosine object with the specified
+   * parameters.
+   *
+   * @param a A double type value for smoothening loss function.
+   *          It must be positive real number, Sharpness of loss
+   *          function is directly propotional to a. It can also
```

Comment:
`proportional`

Reply:
Terribly sorry for the typo. I will fix it.

Reply:
No worries.